client/src/js/directives/bhFindSupplier.js

```diff
@@ -0,0 +1,201 @@
+  * according the name of supplier.
+  *
+  * @return {Array} An array of suppliers
+  */
+  function searchByName(text) {
+    vm.loadStatus = 'loading';
+
+    // format query string parameters
+    var options = {
+      name : text.toLowerCase(),
+      limit : LIMIT
+    };
+
+    return SupplierService.filter(options)
+    .then(function (suppliers) {
+      // loop through each
+      suppliers.forEach(function (supplier) {
+        supplier.label = supplier.name;
+      });
+
+      return suppliers;
+    });
+  }
+
+
+  /**
+  * @method findBy
+  *
+  * @param {object} option The selected option
+  */
+
+  function findBy(key) {
+    vm.selected   = vm.options[key];
+    vm.loadStatus = null;
+    vm.idInput    = undefined;
+    vm.nameInput  = undefined;
+
+    // save the option for later
+    cache.put('optionKey', key);
+  }
+
+  /**
+  * @method reload
+  *
+  * @description This function is responsible for enabling the user to input data
+  * again for search by showing the inputs zones again.
+  */
+  function reload() {
+    vm.showSearchView = true;
+  }
+
+  /**
+  * @method handler
+  *
+  * @param {object} error The error object
+  *
+  * @description This function is responsible for handling errors which occurs
+  * and notify the client into the console
+  */
+  function handler(error) {
+    throw error;
+  }
+
+  /**
+  * @method selectSupplier
+  *
+  * @param {object} supplier The supplier object
+  *
+  * @description This function is responsible for handling the result of the search,
+  * display results and pass the returned supplier to the parent controller
+  */
+
+  function selectSupplier(supplier) {
+    vm.showSearchView = false;
+
+    if (supplier && typeof(supplier) === 'object') {
+      vm.loadStatus = 'loaded';
+      vm.supplier = supplier;
+
+      // call the external function with supplier
+      vm.onSearchComplete({ supplier : supplier });
+
+    } else {
+      vm.loadStatus = 'error';
+    }
+  }
+
+  /**
+  * @method validateNameSearch
+  *
+  * @param {string} value The supplier reference name
+  *
+  * @description Check if the value in the inputs is correct (well defined)
+  */
+  function validateNameSearch(value) {
+    vm.validInput = angular.isDefined(value);
+
+    // Update the nofication
+    if (!vm.validInput) {
+      vm.loadStatus = null;
+    }
+  }
+
+  /**
+  * @method loadDefaultOption
+  *
+  * @param {object} key - the default option key to search by
+  *
+  * @description This function is responsible for changing the option of search.
+  * Search by name
+  */
+  function loadDefaultOption(optionKey) {
+    // change the findBy call
+    findBy(optionKey);
+  }
+
+  /**
+  * @method readInput
+  *
+  * @param {object} event An Event object
+  *
+  * @description This function capture the "Enter" key push of the user and
+  * call a function to do something
+  */
+  function readInput(event) {
+    if (event.keyCode === 13) {
+      submit();
```

Comment:
To prevent submitting the parent form, you should halt the event here.  See `event.preventDefault()`.